src/Reflow/Parser.hs

```diff
@@ -11,11 +13,33 @@ parseFile :: Text -> [Content]
 parseFile t = either (const []) id $ parse parseContent "content" t
 
 parseContent :: Parser [Content]
-parseContent = many (quoted <|> codeBlock <|> normal) <* eof
+parseContent = do
+    h <- option [] (many header)
+    c <- many (quoted <|> codeBlock <|> normal)
+    void eof
+    return (h <> c)
```

Comment:
seems like your style below has been to use `$` after `return` instead of parens